src/coprocessor/gadgets.rs

```diff
@@ -125,7 +125,7 @@ pub(crate) fn construct_list<F: LurkField, CS: ConstraintSystem<F>>(
     cs: &mut CS,
     g: &GlobalAllocator<F>,
     store: &Store<F>,
-    elts: &[&AllocatedPtr<F>],
+    elts: &[AllocatedPtr<F>],
```

Comment:
I think [FG's rant](https://zulip.lurk-lab.com/#narrow/stream/12-rust/topic/PSA.3A.20bugging.20your.20caller.20with.20vector.20allocation/near/154743) can be applied here so we avoid the clones that come with this change

Reply:
I poked at this and couldn't get it to work right away, btw. If you think it's simple and want to take a crack at it, please do.

Reply:
Actually, I figured it out. PR incoming.

Reply:
#1131